### PR TITLE
Compile with deprecation warnings & fix deprecations

### PR DESCRIPTION
**Pull request by magro**
This builds upon #271 to prevent conflicts, please review #271 first.

This PR here lets the compiler warn on deprecations, and fixes deprecations:

* Deprecation of `ToxiproxyContainer.ContainerProxy` - refactors proxied kafka and postgresql containers, also moving them to commons so that they can be shared between the classic and reactive modules
* Replace deprecated `AutowireCapableBeanFactory.applyBeanPostProcessorsAfterInitialization` with `initializeBean`
* Replace deprecated `org.junit.Assert.assertThat` with the hamcrest version

### CODE REVIEW

**Comment by magro** [comment]
Anybody of you up for a review, @hnrkdmsk @danielrehmann?

File: commons/src/testFixtures/java/one/tomorrow/transactionaloutbox/commons/ProxiedPostgreSQLContainer.java [hunk 1]
```diff
@@ -13,19 +13,24 @@
...
-public class ProxiedPostgreSQLContainer extends PostgreSQLContainer<ProxiedPostgreSQLContainer> {
+import static one.tomorrow.transactionaloutbox.commons.ProxiedContainerPorts.findPort;
+import static one.tomorrow.transactionaloutbox.commons.ProxiedContainerSupport.createProxy;
+
+public class ProxiedPostgreSQLContainer extends PostgreSQLContainer<ProxiedPostgreSQLContainer> implements ProxiedContainerSupport {
 
     private static ProxiedPostgreSQLContainer postgres;
     private static ToxiproxyContainer toxiproxy;
-    public static ToxiproxyContainer.ContainerProxy postgresProxy;
+    public static Proxy postgresProxy;
```
Comment (hnrkdmsk) [inline commons/src/testFixtures/java/one/tomorrow/transactionaloutbox/commons/ProxiedPostgreSQLContainer.java:33]: Why do you not use the `ContainerProxy` anymore? As I can see, is the `ContainerProxy` part of the library we're using, but the `Proxy` is coming from somewhere else.

Comment (magro) [reply commons/src/testFixtures/java/one/tomorrow/transactionaloutbox/commons/ProxiedPostgreSQLContainer.java:33]: Well, because it's deprecated :-)

Comment (magro) [reply commons/src/testFixtures/java/one/tomorrow/transactionaloutbox/commons/ProxiedPostgreSQLContainer.java:33]: See e.g.
* https://github.com/Shopify/toxiproxy/issues/470
* https://java.testcontainers.org/modules/toxiproxy/

**Review by danielrehmann — APPROVED** [review]
Thanks for driving this @magro

File: commons/src/testFixtures/java/one/tomorrow/transactionaloutbox/commons/ProxiedPostgreSQLContainer.java (hunk 1 above)
Comment (magro) [reply commons/src/testFixtures/java/one/tomorrow/transactionaloutbox/commons/ProxiedPostgreSQLContainer.java:33]: I'm already marking this as resolved, if there's anything else we can still adjust it.

**Pull request merged**